reactor-core/src/main/java/reactor/core/publisher/Operators.java

```diff
@@ -2310,11 +2325,19 @@ final void drainLoop() {
 		final T value;
 
 		volatile int once;
+		
+		String stepName;
+		
 		ScalarSubscription(CoreSubscriber<? super T> actual, T value) {
+			this(actual, value, null);
+		}
+
+		ScalarSubscription(CoreSubscriber<? super T> actual, T value, String stepName) {
 			this.value = Objects.requireNonNull(value, "value");
 			this.actual = Objects.requireNonNull(actual, "actual");
+			this.stepName = stepName;
```

Comment:
you can `Objects.requireNonNull` here, I prefer the last iteration where the `stepName` was assemble in the constructor.

Reply:
See my second comment. Unnecessary strings in memory. in 99% not used in runtime. I know it is not nice, but runtime is more important I think.